slsim/Deflectors/DeflectorTypes/nfw_hernquist.py

```diff
@@ -26,23 +26,29 @@ def velocity_dispersion(self, cosmo=None):
         :type cosmo: ~astropy.cosmology class
         :return: velocity dispersion [km/s]
         """
+        if "vel_disp" in self._deflector_dict.columns:
+            if self._deflector_dict["vel_disp"] != -1:
+                return self._deflector_dict["vel_disp"]
         # convert radian to arc seconds
-        size_lens_arcsec = self.angular_size_light / constants.arcsec
+        else:
+            size_lens_arcsec = self.angular_size_light / constants.arcsec
 
-        m_halo, c_halo = self.halo_properties
-        # convert angular size to physical size
-        dd = cosmo.angular_diameter_distance(self.redshift).value
-        rs_star = dd * self.angular_size_light
-        vel_disp = vel_disp_composite_model(
-            r=size_lens_arcsec,
-            m_star=self.stellar_mass,
-            rs_star=rs_star,
-            m_halo=m_halo,
-            c_halo=c_halo,
-            cosmo=cosmo,
-            z_lens=self.redshift,
-        )
-        return vel_disp
+            m_halo, c_halo = self.halo_properties
+            m_halo_acc = self._deflector_dict["halo_mass_acc"]
+            m_halo = max(m_halo, m_halo_acc)
+            # convert angular size to physical size
+            dd = cosmo.angular_diameter_distance(self.redshift).value
+            rs_star = dd * self.angular_size_light
+            vel_disp = vel_disp_composite_model(
+                r=size_lens_arcsec,
+                m_star=self.stellar_mass,
+                rs_star=rs_star,
+                m_halo=m_halo,
+                c_halo=c_halo,
+                cosmo=cosmo,
+                z_lens=self.redshift,
+            )
+            return vel_disp
```

Comment:
perhaps add gel_disp to the deflector_dict after calculations happened. Then we would not need to re-compute it if we want to access this number multiple times for the same deflector (just a thought I had and should have done myself before)